src/EditorFeatures/Core.Wpf/InlineErrors/InlineErrorTag.cs

```diff
@@ -0,0 +1,126 @@
+            };
+
+            link.RequestNavigate += HandleRequestNavigate;
+            block.Inlines.Add(link);
+            block.Inlines.Add(": " + _diagnostic.Message);
+            block.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
+            block.Arrange(new Rect(block.DesiredSize));
+
+            var border = new Border
+            {
+                Background = format.BackgroundBrush,
+                Child = block,
+                CornerRadius = new CornerRadius(2),
+                // Highlighting lines are 2px buffer. So shift us up by one from the bottom so we feel centered between them.
+                Margin = new Thickness(0, top: 0, 0, bottom: 2),
+                Padding = new Thickness(2)
+            };
+
+            border.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
+
+            view.ViewportWidthChanged += ViewportWidthChangedHandler;
+            // Need to set these properties to avoid unnecessary reformatting because some dependancy properties
+            // affect layout.
+            // TODO: Not sure if these are needed anymore since the errors are not intratextadornment tags
+            TextOptions.SetTextFormattingMode(border, TextOptions.GetTextFormattingMode(view.VisualElement));
+            TextOptions.SetTextHintingMode(border, TextOptions.GetTextHintingMode(view.VisualElement));
+            TextOptions.SetTextRenderingMode(border, TextOptions.GetTextRenderingMode(view.VisualElement));
+
+            Canvas.SetTop(border, bounds.Bounds.Bottom - border.DesiredSize.Height);
+            Canvas.SetLeft(border, view.ViewportWidth - border.DesiredSize.Width);
+
+            return new GraphicsResult(border,
+                () => view.ViewportWidthChanged -= ViewportWidthChangedHandler);
+
+            void ViewportWidthChangedHandler(object s, EventArgs e)
+            {
+                Canvas.SetLeft(border, view.ViewportWidth - border.DesiredSize.Width);
+            }
+        }
+
+        /// <summary>
+        /// Navigates to the requrest URL
+        /// </summary>
+        private void HandleRequestNavigate(object sender, RoutedEventArgs e)
+        {
+            var link = (Hyperlink)sender;
+            var uri = link.NavigateUri.ToString();
+            Process.Start(uri);
+            e.Handled = true;
+        }
+
+        /// <summary>
+        /// Gets called when the ClassificationFormatMap is changed to update the adornment
+        /// </summary>
+        public void UpdateColor(TextFormattingRunProperties format, UIElement adornment)
+        {
+            var border = (Border)adornment;
+            border.Background = format.BackgroundBrush;
+            var block = (TextBlock)border.Child;
+            block.Foreground = format.ForegroundBrush;
+        }
+
+        public override GraphicsResult GetGraphics(IWpfTextView view, Geometry bounds)
+        {
+            throw new NotImplementedException();
+        }
```

Comment:
this is strange.  i'm not sure the point.  we have two overloads.  but each one throws for the respective subclasses.

Reply:
I'm not sure how else to do this? I need the GetGraphics for Inline Errors to have a TextFormattingRunProperties. Is it better to have only one overload and set the default value for the TextFormattingRunProperties param to null?